src/evict/evict_lru.c

```diff
@@ -448,8 +454,22 @@ __evict_server(WT_SESSION_IMPL *session, bool *did_work)
         return (0);
 
     __wt_epoch(session, &now);
-    if (WT_TIMEDIFF_SEC(now, cache->stuck_time) > WT_MINUTE * 5) {
-#if defined(HAVE_DIAGNOSTIC)
+
+#define WT_CACHE_STUCK_TIMEOUT_SEC 300
+#ifdef HAVE_DIAGNOSTIC
+    /* Enable extra logs 20ms before timing out. */
+    if (!verbose_timeout_flags &&
+      WT_TIMEDIFF_MS(now, cache->stuck_time) >
+        ((WT_CACHE_STUCK_TIMEOUT_SEC + 1) * WT_THOUSAND) - 20) {
```

Comment:
Should this be (WT_CACHE_STUCK_TIMEOUT_SEC * WT_THOUSAND) - 20, i.e.
`diff > 300*1000 - 20`

Reply:
I had to do this because of the use of `WT_TIMEDIFF_SEC`. I have simplified the code in 59c6d96e3c9210038269503d76871ddc8e0781d5.